pkg/preparer/setup.go

```diff
@@ -650,6 +652,71 @@ func getArtifactRegistry(preparerConfig *PreparerConfig) (artifact.Registry, err
 	return artifact.NewRegistry(url, fetcher, osversion.DefaultDetector), nil
 }
 
+func (p *Preparer) BuildRealityAtLaunch() error {
+	// check if reality tree does not exist, if true then reimage occurred and may need
+	// to rebuild reality tree for orphaned pods
+	// a pod is orphaned if they exist on disk but not in intent
+	sub := p.Logger.SubLogger(nil)
+	realityResults, _, err := p.store.ListPods(consul.REALITY_TREE, p.node)
+	if err != nil {
+		sub.WithError(err).Errorln("Could not check reality: %s", err)
+		return err
+	} else if len(realityResults) == 0 {
```

Comment:
nit: since you returned on line 663 you don't really need this `else` and you can write the if on the next line as suggested here: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow